### ISSUE TITLE
Improve usability

### ISSUE THREAD

**Issue by niutech** [matjaztrontelj/PinegrowCommunity#14]
I've tested many Bootstrap designer tools, including Webflow, Jetstrap, Easel, LayoutIt and Pingendo, but Pinegrow is the most full-featured among them. It has the biggest library of Bootstrap components and they could be thoroughly customized. Well done!

However, there are some usability glitches in the UI, which I'm asking you to improve:
- The panels should be draggable, so I could move the Prop panel to the right side and have the Lib panel on the left,
- The left panel should be resizable just as the right panel is,
- When I select an element in the tree on the right, there should be a Delete icon next to the Eye icon (or a Del key binding), so I could remove it easily,
- When I go to the Edit Content mode, the page zooms in, which makes some part of a page hidden behind the right panel,
- The exported HTML code should be formatted, right now it's a single line, e.g. `<body ><div class="jumbotron"><div class="container"><h2 class="text-center">`,
- There should be a public API for pluggable widget libraries, like jQuery UI, jQuery Mobie, Sencha, Dojo, Pure, Foundation, etc.

Apart from that, Pinegrow is a very promising tool!

**Comment by matjaztrontelj** [comment]
Thanks!

Issue "The exported HTML code should be formatted, right now it's a single line, e.g." is fixed. Due to a bug HTML formatting code was not called in version 1.09.

Issue "There should be a public API for pluggable widget libraries" is closed. Pinegrow already has this API: See https://github.com/matjaztrontelj/PinegrowCommunity/wiki#wiki-pinegrow-plugins

Other issues will be taken care of in the upcoming redesign.

**Comment by ialex** [comment]
@matjaztrontelj  i agree whit @niutech you need to work on UX and add better controls to modify the css you can check out webflow as pointed and take some ideas there but you know what you are on the rigth track man so keep doing your amazing job!

**Comment by tmikaeld** [comment]
Sizing can be tricky and always require keyboard typing, to speed that up you could use a stepping knob, dial, arrows.

Example: http://anthonyterrien.com/knob/
For % values this can even be shown visually.
For infinite values, see the infinite dial.

With that, it would be a simple flick of the scrollwheel to change size.

**Comment by Romainpetit** [comment]
@matjaztrontelj I also agree with @niutech, this is for sure the number 1 priority issue. 

The best page builder interface I know is provided by Visual Composer : http://vc.wpbakery.com/

Hope it helps.
